courseTasks/executor/executor.go

```diff
@@ -0,0 +1,103 @@
+package executor
```

Comment:
I feel like the whole `executor` package is overcomplicated. Is there a way to make it easier?